File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/DocumentReference.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Attachment
from base import Period
from base import CodeableConcept
from base import Coding
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class DocumentReferenceContent(BackboneElement):
    format: Optional[Coding] = None
    attachment: Attachment

@dataclass(kw_only=True)
class DocumentReferenceRelatesTo(BackboneElement):
    code: str
    target: Reference

@dataclass(kw_only=True)
class DocumentReferenceContext(BackboneElement):
    event: Optional[List[CodeableConcept]] = field(default_factory=list)
    period: Optional[Period] = None
    related: Optional[List[Reference]] = field(default_factory=list)
    encounter: Optional[List[Reference]] = field(default_factory=list)
    facility_type: Optional[CodeableConcept] = None
    practice_setting: Optional[CodeableConcept] = None
    source_patient_info: Optional[Reference] = None

@dataclass(kw_only=True)
class DocumentReference(DomainResource):
    description: Optional[str] = None
    category: Optional[List[CodeableConcept]] = field(default_factory=list)
    date: Optional[str] = None
    doc_status: Optional[str] = None
    content: list[DocumentReferenceContent] = field(default_factory=list)
    type: Optional[CodeableConcept] = None
    author: Optional[List[Reference]] = field(default_factory=list)
    master_identifier: Optional[Identifier] = None
    custodian: Optional[Reference] = None
    status: str
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    relates_to: Optional[List[DocumentReferenceRelatesTo]] = field(default_factory=list)
    context: Optional[DocumentReferenceContext] = None
    security_label: Optional[List[CodeableConcept]] = field(default_factory=list)
    subject: Optional[Reference] = None
    authenticator: Optional[Reference] = None

File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/DocumentReference.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class DocumentReference : DomainResource, IResource
{
    public string? Description { get; set; }
    public Base.CodeableConcept[]? Category { get; set; }
    public string? Date { get; set; }
    public string? DocStatus { get; set; }
    public required DocumentReference_Content[] Content { get; set; }
    public Base.CodeableConcept? Type { get; set; }
    public Base.ResourceReference[]? Author { get; set; }
    public Base.Identifier? MasterIdentifier { get; set; }
    public Base.ResourceReference? Custodian { get; set; }
    public required string Status { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public DocumentReference_RelatesTo[]? RelatesTo { get; set; }
    public DocumentReference_Context? Context { get; set; }
    public Base.CodeableConcept[]? SecurityLabel { get; set; }
    public Base.ResourceReference? Subject { get; set; }
    public Base.ResourceReference? Authenticator { get; set; }

    public class DocumentReference_Content : BackboneElement
    {
        public Base.Coding? Format { get; set; }
        public required Base.Attachment Attachment { get; set; }
    }

    public class DocumentReference_RelatesTo : BackboneElement
    {
        public required string Code { get; set; }
        public required Base.ResourceReference Target { get; set; }
    }

    public class DocumentReference_Context : BackboneElement
    {
        public Base.CodeableConcept[]? Event { get; set; }
        public Base.Period? Period { get; set; }
        public Base.ResourceReference[]? Related { get; set; }
        public Base.ResourceReference[]? Encounter { get; set; }
        public Base.CodeableConcept? FacilityType { get; set; }
        public Base.CodeableConcept? PracticeSetting { get; set; }
        public Base.ResourceReference? SourcePatientInfo { get; set; }
    }
}